s2/bq/doc/REFERENCE.md

```diff
@@ -0,0 +1,162 @@
+## Reference
+
+### S2
+
+#### VERSION
+
+{{% bannerNote type="code" %}}
+s2.VERSION()
+{{%/ bannerNote %}}
+
+Returns the current version of the s2 library.
+
+#### GEOJSONBOUNDARY_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMKEY(key STRING) 
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners of a quadkey.
+
+* `key`: `STRING` quadkey to extract the corners from.
+
+#### GEOJSONBOUNDARY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners given a longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### ID_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Convert from hilbert quadtree id to s2 cell id.
+
+* `key`: `STRING` quadkey to be converted.
+
+#### ID_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the s2 cell id for a given longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### KEY_FROMID
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Convert from s2 cell id to hilbert quadtree id.
+
+* `id`: `INT64` s2 cell id to be converted.
+
+#### KEY_FROMLONGLAT
```

Comment:
I'm having issues understanding what's a `key` and what's an `id`. What's the deal with it? Can't we use the 64-bit [cell id](https://s2geometry.io/devguide/s2cell_hierarchy.html#s2cellid-numbering)? That should uniquely identify the cell ids and we can use that binary representation in an INT64 right?

Reply:
With key we mean HILBERTQUADKEY, example, '4/032212303102210' and with id we mean the S2 cell ID which is an INT64 as you say. Yes, both can uniquely identify the cell, so to simplify the module I think it's a good idea to do what you say. This is a similar situation to our quadkey/quadint module, so I will keep also the parsers as we do with the quadkey module.